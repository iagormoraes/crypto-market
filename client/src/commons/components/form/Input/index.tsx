import { HTMLAttributes } from 'react';
import classnames from 'classnames';
import { FieldInputProps } from 'formik';

interface Props extends HTMLAttributes<HTMLInputElement> {
  field: FieldInputProps<string>;
  form: string;
}

export default function Input({ field, form, ...props }: Props) {
  return (
    <div className="flex flex-col mb-4">
      <label className="text-white capitalize text-lg mb-2">{field.name}</label>
      <input
        {...field}
        {...props}
        className={classnames(
          'px-2 py-4 rounded font-bold text-gray-600 text-xl',
          props.className,
        )}
      />
    </div>
  );
}
