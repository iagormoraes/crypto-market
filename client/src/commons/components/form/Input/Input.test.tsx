import Input from './index';
import { fireEvent, render, screen } from '@testing-library/react';

describe('<Input/>', () => {
  const build = () => ({
    value: 'value',
    name: 'name',
    onChange: jest.fn(),
    onBlur: jest.fn(),
  });

  let props = build();

  beforeEach(() => {
    props = build();
  });

  it('matches snapshot', () => {
    const { container } = render(<Input form="input" field={props} />);

    expect(container).toMatchSnapshot();
  });

  it('triggers onChange event', () => {
    render(<Input form="input" field={props} />);

    const input = screen.getByLabelText(props.name) as HTMLInputElement;

    fireEvent.change(input, { target: { value: '' } });

    expect(props.onChange).toHaveBeenCalled();
    expect(input.value).toBe(props.value);
  });

  it('triggers onBlur event', () => {
    render(<Input form="input" field={props} />);

    const input = screen.getByLabelText(props.name);

    fireEvent.blur(input);

    expect(props.onBlur).toHaveBeenCalled();
  });
});
