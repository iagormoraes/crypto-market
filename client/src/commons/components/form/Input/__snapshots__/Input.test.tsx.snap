// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input/> matches snapshot 1`] = `
<div>
  <div
    class="flex flex-col mb-4"
  >
    <label
      class="text-white capitalize text-lg mb-2"
      for="name"
    >
      name
    </label>
    <input
      class="px-2 py-4 rounded font-bold text-gray-600 text-xl"
      data-testid="input"
      id="name"
      name="name"
      value="value"
    />
  </div>
</div>
`;
